### DISCUSSION TITLE
BAPI_DOCUMENT_CREATE2 fails

### DISCUSSION THREAD

**Discussion by BerniUhr** [SAP/PyRFC#236] [Q&A] [answered]
Hi,

would like to create a new document and check-in a local file. For that I'm calling the
following RFC sequence:
- BAPI_DOCUMENT_CREATE2
- BAPI_TRANSMISSION_COMMIT

This works to create a document only. But if I try to check a file additionally with BAPI_DOCUMENT_CREATE2
I get the following error:
BAPI_DOCUMENT_CREATE2: RFC_NOT_FOUND (rc=17): key=RFC_NOT_FOUND, message=Function RFC_START_PROGRAM not found [MSG: class=, type=, number=, v1-4:=;;;]

Does it has to do with with missing callback functionality in the pyrfc library? Better with another library?
Should I use other BAPI functions to do this?
Do I have to use it in another way?

If I can use pyrfc for that, an example would be great.

Best regards
Bernhard

**Comment by bsrdjan** [comment, marked as answer]
Hello Bernhard,

RFC callback functionality is not supported any more, no matter which library (Python, Node, Java ...)

Whether to use another BAPI is hard to say, it depends on scenario, system release, requirements, etc. but there is nothing wrong with using `BAPI_DOCUMENT_CREATE2` either.

After brief investigation, you could try calling the BAPI as described here:

- https://wiki.scn.sap.com/wiki/display/PLM/Calling+DMS+BAPIs+from+external+RFC+clients+without+using+CALLBACK+mechanism

- https://blogs.sap.com/2013/11/25/create-an-sap-dms-document-attaching-a-file-from-the-application-server-and-checking-in-an-ftp-vault-with-automatic-filename-creation/

If you have access to ABAP dev environment, you can you call BAPI also from ABAP SE37 transaction or from ABAP report, to verify parameters and check if error related to Python/PyRFC. It does not seem to be the case, looks more like the right parameter combination needed to call this particular BAPI

Kind regards, Srdjan

**Reply by BerniUhr** [reply]
Hello Srdjan,

thank you for your response. The point was to use 
PF_FTP_DEST='SAPFTPA', PF_HTTP_DEST='SAPHTTPA'
additionally.
Now it works. But to understand it. Does it has to be exactly these names 'SAPFTPA' and 'SAPHTTPA' or are these random names?
And isn't there a complete and good API Description for BAPI_DOCUMENT_CREATE2. Other than in transaction BAPI?

Best regards
Bernhard

**Comment by bsrdjan** [comment]
Hello Bernhard,

this repository supports BAPIs and FMs consumption in general. There are many BAPIs and I am not familiar with usage details for all of them. For questions about BAPI_DOCUMENT_CREATE2 and other Document Management BAPIs, I would suggest to check with respective DMS experts and consultants. Sorry that I can't help with PF_FTP_DEST and PF_HTTP_DEST parameters either. 

Kind regards,
Srdjan
